src/modules/ERC4337.sol

```diff
@@ -0,0 +1,64 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity ^0.8.18;
+
+import { Calls } from "./Calls.sol";
+import { IAccount, PackedUserOperation } from "./interfaces/IAccount.sol";
+import { IERC1271, IERC1271_MAGIC_VALUE_HASH } from "./interfaces/IERC1271.sol";
+import { Entrypoint } from "./interfaces/IEntrypoint.sol";
+
+abstract contract ERC4337 is IAccount, Calls {
+
+  uint256 internal constant SIG_VALIDATION_FAILED = 1;
+
+  address public immutable entrypoint;
```

Comment:
Should we have a self callable update entrypoint function instead of being immutable?

Reply:
The entrypoint lives in the implementation, it cannot be made mutable or else the proxy wallets can't access it (or need to set it up during init, and we do not have init)